crates/router/src/routes/routing.rs

```diff
@@ -565,17 +565,8 @@ pub async fn routing_retrieve_default_config(
         &req,
         (),
         |state, auth: auth::AuthenticationData, _, _| {
-            routing::retrieve_default_routing_config(state, auth.merchant_account, transaction_type)
+            routing::retrieve_default_routing_config(state, auth.profile_id, transaction_type)
```

Comment:
Also pass `auth.merchant_account`. It will be used if `profile_id` is not present.